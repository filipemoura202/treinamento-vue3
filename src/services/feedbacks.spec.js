import FeedBackService from './feedbacks'
import mockAxios from 'axios'

jest.mock('axios')

describe('Feedback Service', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('Should return all feedbacks per type', async () => {
    const resp = {
      pagination: {},
      results: []

    }

    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ data: resp })
    })

    const response = await FeedBackService(mockAxios).getAll({
      limit: 2,
      offset: 0
    })
    expect(response.data).toHaveProperty('pagination')
    expect(response.data).toHaveProperty('results')
    expect(response).toMatchSnapshot()
  })

  it('Should return a summary of feedbacks', async () => {
    const summary = {
      all: 7,
      idea: 3,
      issue: 3,
      other: 0
    }

    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ data: summary })
    })

    const response = await FeedBackService(mockAxios).getSummary()
    expect(response.data).toHaveProperty('all')
    expect(response.data).toHaveProperty('idea')
    expect(response.data).toHaveProperty('issue')
    expect(response.data).toHaveProperty('other')
    expect(response).toMatchSnapshot()
  })
})
