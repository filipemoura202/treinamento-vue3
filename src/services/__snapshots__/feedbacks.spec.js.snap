// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Service Should return a summary of feedbacks 1`] = `
Object {
  "data": Object {
    "all": 7,
    "idea": 3,
    "issue": 3,
    "other": 0,
  },
}
`;

exports[`Feedback Service Should return all feedbacks per type 1`] = `
Object {
  "data": Object {
    "pagination": Object {},
    "results": Array [],
  },
}
`;
